pandas/core/generic.py

```diff
@@ -230,6 +230,15 @@ def _init_mgr(cls, mgr, axes=None, dtype=None, copy=False):
                 mgr = mgr.astype(dtype=dtype)
         return mgr
 
+    @classmethod
+    def _from_mgr(cls, mgr: BlockManager):
+        """
+        Fastpath constructor for if we have only a BlockManager.
+        """
+        obj = object.__new__(cls)
```

Comment:
might consider calling __finalize__ here (have to pass the method as an arg)

Reply:
one of the use cases i have in mind is in groupby `_chop` where we are potentially creating many many Series/DataFrame objects, hoping we can avoid the extra call